tools/curl/gurl/__init__.py

```diff
@@ -13,6 +14,9 @@ def _parse_raw_http(parser_builder, raw_http, parser_extract=None):
     parser = parser_builder(callbacks)
     parser.feed_data(raw_http)
     callbacks.data["version"] = parser.get_http_version()
+    if "body" in callbacks.data:
+        # body must be base64
+        callbacks.data["body"] = base64.encodebytes(callbacks.data["body"]).decode("utf-8")
```

Comment:
Why not this?
```python
base64.encode(callbacks.data["body"])
```

Reply:
because json serializer throw me an invalid character if I don't do it that way.

Reply:
Sorry, I meant `base64.b64encode`.

Reply:
I supose, I try both, same result, JSON parsing error

Reply:
Weird, could you recreate it now?

Reply:
File "/usr/lib64/python3.8/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/usr/lib64/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib64/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable

Reply:
./gcurl www.google.com

Reply:
maybe related with RequestResponseCallbacks and how put multiplart body together

Reply:
You are absolutely right, my mistake.